### ISSUE TITLE
Add action creator for `history.block`

### ISSUE THREAD

**Issue by tesujimath** [FormidableLabs/redux-little-router#124]
When using v4 of mjackson's history library, I can use history.block() to confirm navigation.  However, redux-little-router is using v3, which doesn't have this.

If I use the v4 history library directly, redux-little-router doesn't know what's going on with the route changes.  So I seem to be stuck until redux-little-router upgrades its history library to v4, and provides a way to access it from application code.

Any suggestions?

**Comment by tesujimath** [comment]
Since I need this, I propose to fix it myself and submit a PR.  Here's what I'm thinking:

1. Update history library used by redux-little-router to v4.

2. Expose the history instance used by the router to the application, so that route changes may be initiated by the application calling history.push(), rather than dispatching an action.  They can therefore be subject to user confirmation via the history.block() mechanism, and if not confirmed, the route would not change, and no action would get dispatched by the router.

Caveat: I haven't looked into this in any detail yet.

How does that sound?

**Comment by divmain** [comment]
@tesujimath, you may have missed that `history` is now at `4.4.1`.

**Comment by tesujimath** [comment]
So that's 1 already done, easy.  ;-)

Any comments on 2, which is really the main point, isn't it?

**Comment by divmain** [comment]
I don't have any objection, although @tptee may have an alternative in mind.  @tptee?

**Comment by tesujimath** [comment]
Hello @tptee, would really appreciate some feedback on this idea.  Exposing the history library to the application seems like the simplest solution.  But I'd rather not end up having to maintain my own fork, in case you have a different idea in mind.

**Comment by tptee** [comment]
Now that we have official action creators wrapping the `history` API, think it's best to add a new action creator for `history.block` rather than exposing `history entirely.

**Comment by tesujimath** [comment]
I no longer have an opinion about this, since I switched to router5 a while ago, sorry.

**Comment by opavader** [comment]
@tptee This will be a really useful feature to add.
I will be happy to come with a PR. Any heads up on you will like to have it implemented?

**Comment by tptee** [comment]
Check out the `block` and `unblock` action creators in `v14.3.0`!
